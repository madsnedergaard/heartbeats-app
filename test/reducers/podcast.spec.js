import podcastAppReducer from '../../app/podcasts/reducer';
import { TOGGLE_AUDIO,SELECT_SERIES,SELECT_EPISODE,FETCHING_EPISODES,FETCHING_EPISODES_FAILED,FETCHING_EPISODES_SUCCEEDED,FETCHING_SERIES,FETCHING_SERIES_FAILED,FETCHING_SERIES_SUCCEEDED,SET_FETCHING_STATUS } from '../../app/podcasts/actions';

describe('Podcast reducer', () => {
  describe('player', () => {
    it('should toggle audio', () => {
      expect(podcastAppReducer(undefined, { type: TOGGLE_AUDIO })).toMatchSnapshot();
    });

    it('should select a series', () => {
      expect(podcastAppReducer(undefined, { type: SELECT_SERIES, selectedShow: 1 })).toMatchSnapshot();
    });

    it('should select episode and start playing', () => {
      expect(podcastAppReducer(undefined, { type: SELECT_EPISODE, selectedEpisode: 1, selectedSource: 'source', isPlaying: true })).toMatchSnapshot();
    });
  });
  describe('API', () => {
    it('should start fetching series', () => {
      expect(podcastAppReducer(undefined, { type: FETCHING_SERIES })).toMatchSnapshot();
    });
    // TODO Mock API calls
  })

  it('should handle unknown action type', () => {
    expect(podcastAppReducer(undefined, { type: 'unknown' })).toMatchSnapshot();
  });
})
