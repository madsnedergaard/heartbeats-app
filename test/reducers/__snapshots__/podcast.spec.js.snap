// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Podcast reducer API should start fetching series 1`] = `
Object {
  "isFetching": true,
  "isPlaying": false,
  "online": false,
  "podcasts": Array [],
  "selectedEpisode": null,
  "selectedShow": null,
  "selectedSource": null,
  "series": Array [],
  "time": 0,
}
`;

exports[`Podcast reducer player should select a series 1`] = `
Object {
  "isFetching": false,
  "isPlaying": false,
  "online": false,
  "podcasts": Array [],
  "selectedEpisode": null,
  "selectedShow": 1,
  "selectedSource": null,
  "series": Array [],
  "time": 0,
}
`;

exports[`Podcast reducer player should select episode and start playing 1`] = `
Object {
  "isFetching": false,
  "isPlaying": true,
  "online": false,
  "podcasts": Array [],
  "selectedEpisode": 1,
  "selectedShow": null,
  "selectedSource": "source",
  "series": Array [],
  "time": 0,
}
`;

exports[`Podcast reducer player should toggle audio 1`] = `
Object {
  "isFetching": false,
  "isPlaying": true,
  "online": false,
  "podcasts": Array [],
  "selectedEpisode": null,
  "selectedShow": null,
  "selectedSource": null,
  "series": Array [],
  "time": 0,
}
`;

exports[`Podcast reducer should handle unknown action type 1`] = `
Object {
  "isFetching": false,
  "isPlaying": false,
  "online": false,
  "podcasts": Array [],
  "selectedEpisode": null,
  "selectedShow": null,
  "selectedSource": null,
  "series": Array [],
  "time": 0,
}
`;
